### DISCUSSION TITLE
Summaries extension - Other Month - Opens wrong month.

### DISCUSSION THREAD

**Discussion by chrismalek** [NotePlan/plugins#459] [Q&A]
I am not sure if this is a bug so I am first posting it to the discussions.

The "Summaries" extension. I am trying to go back and do "periodStat" for a few months ago.

- `/periodStats`
- Choose "Other Month"
- Year = 2023
- Month = 4

The result is that it is opening up May instead of April. Even if I run the `/periodStats` command in the 2023-04 note the 2023-05 opens up and that is updated with the April data. 

This looks like an "off by one" bug.

**Comment by chrismalek** [comment]
I got around this by just opening the "2023-04" monthly note, then telling it to append stats to the open note.

**Reply by dwertheimer** [reply]
@jgclark

**Comment by jgclark** [comment]
@chrismalek I've just had a look and I'm confused: your scenario works for me; the code looks OK, and hasn't changed in over a year.
However, the code is using NP's own date API and I been transitioning away from to an opensource one that has better timezone handling and that's had vastly more testing.
So I'll re-write this part, and then send you a build to test if I may (in Discord)?

**Comment by chrismalek** [comment]
@jgclark  I am not on discord. It gives me the heeby-jeebies :-) I am old and chat apps make me anxious and fills me with too much noise.  Can you ping me on GitHub and I will test it.

**Comment by jgclark** [comment]
@chrismalek OK, here's an updated version to test, please. I've re-written the time period calculation codes for months, quarters and years, so please test as many as you can.
Unzip into your NotePlan `Plugins/jgclark.Summaries` folder. It will keep all previous settings. To revert to the previous version use the Plugin Preferences pane to delete and then reinstall (which goes to the published version from GH).
[jgclark.Summaries.0.19.2.zip](https://github.com/NotePlan/plugins/files/12182829/jgclark.Summaries.0.19.2.zip)

**Comment by chrismalek** [comment]
Testing successful for what I am using this for which is limited to period stats into Monthly notes.   I put my tags on the daily notes then push the summary to the daily notes.




- I was in a random daily note:
- `/periodStats`
- Choose "Other Month"
- Year = 2023
- Month = 4

When I chose to output in April 2023 it did append into that month.

This is what I was reporting was off by one earlier.

I also tested 

- `/periodStatus`
- This Month
- Append into 2023-07 monthly note.

It worked fine and appended to the correct place.

👌 🫡

@jgclark

**Comment by jgclark** [comment]
Thanks, @chrismalek. I'll release it soon.
